custom-routes-vis/src/main/java/com/google/sps/servlets/GetDirectionsServlet.java

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.HttpURLConnection;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/get-directions")
+public class GetDirectionsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String origin = request.getParameter("origin");
+    String destination = request.getParameter("destination");
+    String endpoint = request.getParameter("endpoint");
+
+    String apiKey = request.getParameter("apiKey");
+    String directionsApiKey = System.getenv("DIRECTIONS_API_KEY");
+    String key = apiKey == "" ? apiKey : directionsApiKey;  // This works but it seems backwards to me.
+
+    URLConnection connection;
+
+    switch (endpoint) {
+      case "compute-routes":
+        connection = sendPostRequestToRoutesApi(origin, destination, false, key);
+        break;
+      case "compute-routes-alpha":
+        connection = sendPostRequestToRoutesApi(origin, destination, true, key);
+        break;
+      default:  // Default to use the Directions API.
+        connection = sendGetRequestToDirectionsApi(origin, destination, key);
+        break;
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(readResponse(connection));
+  }
+
+  /** 
+   * Sends request to the Directions API and returns the connection.
+   * @param origin Origin of the requested routes in "lat,lng" format.
+   * @param destination Destination of the requested routes in "lat,lng" format.
+   * @param key The API key.
+   */
+  public URLConnection sendGetRequestToDirectionsApi(String origin, String destination, String key) 
+      throws IOException, MalformedURLException {
+    String originDestination = "&origin=" + origin + "&destination=" + destination;
```

Comment:
Rename this variable `originDestination` doesn't sound too nice

Reply:
Do you really need this variable? You only use it once. Also seems strange because you handle its separately to the key variable even through its the same formatting.

Reply:
You are right I don't need it. It just happened during a lot of moving around